Feat : integrate eventbridge for aws sdk v3

guillaumeduboc opened the pull request:
Fix #572

guillaumeduboc committed:
fix(aws-sdk-v3): only call once the wrapped function

guillaumeduboc committed:
feat(aws-eventbridge-sdk-v3): support eventbridge v3

guillaumeduboc commented:
The promise returned by the aws-sdk v3 wrapper, was not the promise responsible for adding the event to the trace. 
Hence, adding the event to the trace was not awaited and the trace was often incomplete (`premature_exit: true`).

Furthermore, the wrapped function was called twice, resulting in 2 events published on the bus instead of one.


This bug is fixed in my first commit

nemoshlag committed:
Merge branch 'master' into feat/aws-sdk-v3/eventbridge

galbash commented on src/events/aws_sdk_v3.js:
```diff
@@ -324,6 +373,7 @@ function getOperationByCommand(command) {
  */
 function AWSSDKv3Wrapper(wrappedFunction) {
     return function internalAWSSDKv3Wrapper(command) {
+        let responsePromise = wrappedFunction.apply(this, [command]);
```
I this this may lead to us calling the function twice for unsupported services

guillaumeduboc replied:
Nice spot, I've fixed it in the next commit

guillaumeduboc committed:
fix(aws-sdk-v3): duplicate call to wrapped function

Remove duplicate call to wrapped function for unsupported resources of the aws sdk v3
